bot/exts/moderation/slowmode.py

```diff
@@ -66,37 +89,98 @@ async def set_slowmode(
         humanized_delay = time.humanize_delta(delay)
 
         # Ensure the delay is within discord's limits
-        if slowmode_delay <= SLOWMODE_MAX_DELAY:
-            log.info(f"{ctx.author} set the slowmode delay for #{channel} to {humanized_delay}.")
-
-            await channel.edit(slowmode_delay=slowmode_delay)
-            if channel.id in COMMONLY_SLOWMODED_CHANNELS:
-                log.info(f"Recording slowmode change in stats for {channel.name}.")
-                self.bot.stats.gauge(f"slowmode.{COMMONLY_SLOWMODED_CHANNELS[channel.id]}", slowmode_delay)
+        if not slowmode_delay <= SLOWMODE_MAX_DELAY:
+            log.info(
+                f"{ctx.author} tried to set the slowmode delay of #{channel} to {humanized_delay}, "
+                "which is not between 0 and 6 hours."
+            )
 
             await ctx.send(
-                f"{Emojis.check_mark} The slowmode delay for {channel.mention} is now {humanized_delay}."
+                f"{Emojis.cross_mark} The slowmode delay must be between 0 and 6 hours."
             )
+            return
 
-        else:
+        if duration is not None:
+            slowmode_duration = time.relativedelta_to_timedelta(duration).total_seconds()
+            humanized_duration = time.humanize_delta(duration)
+
+            expiration_time = datetime.now(tz=UTC) + timedelta(seconds=slowmode_duration)
+            expiration_timestamp = discord_timestamp(expiration_time, TimestampFormats.RELATIVE)
+
+            # Only update original_slowmode_cache if the last slowmode was not temporary.
+            if not await self.slowmode_expiration_cache.contains(channel.id):
+                await self.original_slowmode_cache.set(channel.id, channel.slowmode_delay)
+            await self.slowmode_expiration_cache.set(channel.id, expiration_time.timestamp())
+
+            self.scheduler.schedule_at(expiration_time, channel.id, self._revert_slowmode(channel.id))
```

Comment:
The task ID has to be unique. If you're scheduling with a channel ID that already has a schedule, this creates a clash and the new coroutine won't be registered.

https://github.com/python-discord/bot-core/blob/16835478af6b842d776d634e53aab8ee31bd350c/pydis_core/utils/scheduling.py#L65-L66

You're already checking whether the current slowmode is scheduled to expire just above, can cancel the current task while you're at it (although this logic will probably change if you're changing to a single cache).